### PR TITLE
feat(omni): auto-discover available machines when neither machineClass nor machines is set

### PR DESCRIPTION
**Pull request by devantler**
When using `Talos x Omni`, if neither `machineClass` nor `machines` is configured, KSail now automatically queries Omni for available (unallocated) machines and uses them for node allocation — instead of returning `ErrMachineAllocationRequired`.

This adds a `ListAvailableMachines` method to the Omni provider that queries `MachineStatus` resources with the `MachineStatusLabelAvailable` label. At the provisioner level, a new `resolveOmniMachines` step runs before template building: when neither allocation mode is configured, it discovers the required number of available machines and passes their UUIDs as static `Machines` in the cluster template.

## Type of change

- [x] 🚀 New feature

### CODE REVIEW

File: pkg/svc/provisioner/cluster/talos/provisioner_omni.go [hunk 1]
```diff
@@ -216,6 +223,42 @@ func (p *Provisioner) omniMachines() []string {
...
+	omniProv *omniprovider.Provider,
+) ([]string, error) {
+	if p.omniMachineClass() != "" {
+		return nil, nil
+	}
+
+	if machines := p.omniMachines(); len(machines) > 0 {
+		return machines, nil
+	}
+
```
[the remark this answers is not in the document]
Comment (devantler) [reply pkg/svc/provisioner/cluster/talos/provisioner_omni.go]: Fixed. resolveOmniMachines now detects when both machineClass and machines are set and returns ErrMachineAllocationConflict early. Added a test for this case.

File: pkg/svc/provider/omni/provider.go [hunk 2]
```diff
@@ -395,3 +395,41 @@ func (p *Provider) LatestTalosVersion(ctx context.Context) (string, []string, er
...
+		ctx,
+		p.st,
+		state.WithLabelQuery(resource.LabelExists(omnires.MachineStatusLabelAvailable)),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list available machines: %w", err)
+	}
+
+	ids := make([]string, 0, min(count, machines.Len()))
+
```
[the remark this answers is not in the document]
Comment (devantler) [reply pkg/svc/provider/omni/provider.go:426]: Fixed. ListAvailableMachines now validates count at the start: returns an error for negative values and an empty slice for zero. Added tests for both edge cases.

File: pkg/svc/provider/omni/provider.go [hunk 3]
```diff
@@ -395,3 +395,50 @@ func (p *Provider) LatestTalosVersion(ctx context.Context) (string, []string, er
 func (p *Provider) IsAvailable() bool {
 	return p.st != nil
 }
+
+// ListAvailableMachines queries Omni for machines that are available (not allocated
+// to any cluster) and returns up to count machine UUIDs.
```
[the remark this answers is not in the document]
Comment (devantler) [reply pkg/svc/provider/omni/provider.go]: Fixed. Updated the doc comment to say "exactly count" instead of "up to count".

File: pkg/svc/provider/omni/provider.go [hunk 4]
```diff
@@ -395,3 +395,50 @@ func (p *Provider) LatestTalosVersion(ctx context.Context) (string, []string, er
...
+// Returns ErrInsufficientAvailableMachines when fewer than count machines are available.
+// Returns an error if count is negative, and an empty slice if count is zero.
+func (p *Provider) ListAvailableMachines(ctx context.Context, count int) ([]string, error) {
+	if p.st == nil {
+		return nil, provider.ErrProviderUnavailable
+	}
+
+	if count < 0 {
+		return nil, fmt.Errorf("%w: count must not be negative (%d)", ErrInsufficientAvailableMachines, count)
```
[the remark this answers is not in the document]
Comment (devantler) [reply pkg/svc/provider/omni/provider.go]: Good point. Changed to a plain fmt.Errorf without wrapping ErrInsufficientAvailableMachines, so callers can distinguish invalid input from insufficient capacity via errors.Is.

File: pkg/svc/provider/omni/provider_test.go [hunk 5]
```diff
@@ -275,3 +275,133 @@ func TestDeleteNodes_ClusterExists_RemovesCluster(t *testing.T) {
...
+
+func TestListAvailableMachines_NegativeCount(t *testing.T) {
+	t.Parallel()
+
+	prov := omni.NewProviderWithState(newInMemState())
+
+	machines, err := prov.ListAvailableMachines(context.Background(), -1)
+
+	require.Error(t, err)
+	require.ErrorIs(t, err, omni.ErrInsufficientAvailableMachines)
```
[the remark this answers is not in the document]
Comment (devantler) [reply pkg/svc/provider/omni/provider_test.go]: Updated the test to assert NotErrorIs for ErrInsufficientAvailableMachines and check the error message instead.
